Makefile.am

```diff
@@ -2,6 +2,10 @@ AUTOMAKE_OPTIONS=foreign
 
 AM_CFLAGS=-pedantic -Wall -Werror -Wextra
 
+if LINUX
```

Comment:
Could defining `_XOPEN_SOURCE` on non-linux platforms cause problems? If not, what do you think of always defining it? That way we could avoid the complexities of testing for the host os.